### PR TITLE
temporal deploy of cartography

### PR DESCRIPTION
**Pull request by Mandeep3838**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,3 @@
+NEO4J_PASSWORD=neo4j
```
Comment (ashwin-borneo) [inline .env:1]: no passwords in plain text. can we switch this to a key that needs to be passed? or better yet it needs to be credential/IAM role-based access

File: src/workflows/workflows.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+// Only import the activity types
+import type * as activities from '../activities/cartography';
+import { CartographyResult } from '../activities/cartography/types'
+
+const { cartographyRun } = proxyActivities<typeof activities>({
+  startToCloseTimeout: '1 hour',
+  taskQueue: 'cartography-wf-activity'
+});
+
+export async function runSync(): Promise<CartographyResult> {
```
Comment (ashwin-borneo) [inline src/workflows/workflows.ts:11]: change runSync to runCartographySync ? it's more understandable

File: src/activities/cartography/activities.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+var options = {
+  mode: 'text',
+  args: ['--neo4j-uri', 'bolt://localhost:7687', '--neo4j-user', 'neo4j', '--neo4j-password-env-var', 'NEO4J_PASSWORD', '--permission-relationships-file', 'cartography/permissions_mapping_file.yaml', '--aws-best-effort-mode', '--aws-requested-syncs', 'dynamodb,s3_list_only']
+};
+
+export async function cartographyRun(): Promise<CartographyResult> {
+  console.log("reached the execution stage")
+  try {
+    const result = await pythonShellRun('cartography/__main__.py', options)
```
Comment (ashwin-borneo) [inline src/activities/cartography/activities.ts:16]: Why are we doing. a Shell execution? Why aren't we directly calling the Python function along with the args?

Comment (ongteckwu) [reply src/activities/cartography/activities.ts:16]: You mean we should write the activity in Python?

Comment (ashwin-borneo) [reply src/activities/cartography/activities.ts:16]: nope. you don't need to execute it as a Shell. You can add temporal/worker package in the Cartography and expose the function you intend to call as activity. then register that activity as a ProxyActivity. check the repo borneo-temporal-samples

Comment (Mandeep3838) [reply src/activities/cartography/activities.ts:16]: Cartography is built as a cli tool in python. The end return is a sys.exit for a cartography main function.
If we try to wrap it inside a function, then we need to change the parser function of the cli to pass the arguments.
Also, may need changes in the system return error signals in all cartography package
Currently in the cli there are:
STATUS_SUCCESS = 0
STATUS_FAILURE = 1
STATUS_KEYBOARD_INTERRUPT = 130
What are the drawbacks if we follow this approach?

File: .env (hunk 1 above)
Comment (Mandeep3838) [reply .env:1]: It is just for testing.
Env variable will be set in the pod using deployment templates

**Review by ongteckwu — APPROVED** [review]
LGTM

**Pull request merged**